object-cache.php

```diff
@@ -340,6 +344,16 @@ class WP_Object_Cache {
 	const USE_GROUPS = WP_REDIS_USE_CACHE_GROUPS;
 
 	/**
+	 * Whether to use Redis::pconnect vs Redis::connect.
```

Comment:
We are using this plugin with Microsoft Azure, based on their suggestion, and they (Azure support) requested that we switch to using persistent connections with Redis (the exact message is below). And despite that, since this is a feature of phpredis, IMO, it definitely makes sense to allow WordPress developers to take advantage of that feature.

> During the testing we discovered that a majority of issues were caused by requests which were awaiting a response from the external dependency; this includes both MySQL and Redis Cache.
> 
> Essentially, when the requests start coming into the application, WordPress does its thing going out to the MySQL database for information, then using the Object-Cache, stores the appropriate information in the Cache for future use.
> 
> At some point, usually around the first ~100 requests, we start to see requests begin to Queue. Once the Web Apps start to Queue, the application doesn’t seem to be able to recover from this state due to open connections to the Database (stuck in FINWAIT) then we get the resulting HTTP 5xx exceptions when the HTTP Queue has reached its max limit.
> 
> In this scenario when resources are starting to hang (such as the connections to MySQL) we start to risk port exhaustion, which the best work around is to enable persistent connections as this will reuse a single connection to the external resources. We then moved to reduce the number of ports which were used by changing the wp-redis to use a persistent connection.
> 
> As a result, we saw an increase in the number of requests per second which the application was able to handle, but unfortunately, we still ran into the issues with MySQL which left a number of open ports with the eventual result of the HTTP Queue filling to the max queue length and the same HTTP 5xx exceptions we saw in the past.
> 
> The recommendations that we have in regards to these issues would be as follows:
> 
> Use Persistent connections for MySQL
> Use Persistent connections for Redis